src/core/search/vector_utils.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/vector_utils.h"
+
+#include <cmath>
+#include <memory>
+
+#include "base/logging.h"
+
+namespace dfly::search {
+
+using namespace std;
+
+namespace {
+
+// Euclidean vector distance: sqrt( sum: (u[i] - v[i])^2  )
+__attribute__((optimize("fast-math"))) float L2Distance(const float* u, const float* v,
```

Comment:
I just didn't bother to do it. This actually complies down to pretty efficient vectorized instructions.

We can check for nan when indexing. Other than that we can try see what error we can accumulate at max.

See, knn queries usually care only about correct ordering. I think the probability that even two entires swap because of error is very low. 

Anyways, we can investigate it in the future but its not worth the effort now

Reply:
1. Its 95% of computing time on most queries
2. Yes, I was surprised how good the compiler is. Its not slower than manual avx implementations from hsnwlib that I tried

Reply:
Wow nice, I am covered. @royjacobson 400%? NICE